src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/CollectionDataContract.cs

```diff
@@ -68,14 +68,16 @@ public T Value
 
     [DataContract(Namespace = "http://schemas.microsoft.com/2003/10/Serialization/Arrays")]
 #if USE_REFEMIT
-    public struct KeyValue<K, V>
+    public class KeyValue<K, V>
 #else
```

Comment:
> I'm concerned on this as it diverges from desktop .NET.

If the change proves to be beneficial to desktop, we should port the fix to desktop too.